server/src/main/java/org/jboss/as/server/ServerService.java

```diff
@@ -461,6 +455,12 @@ protected void finishBoot(boolean suspend) throws ConfigurationPersistenceExcept
 
     @Override
     protected void postBoot() {
+        String message = "";
+        if (configuration.getServerEnvironment().getServerConfigurationFile() != null) {
+            String serverConfig = configuration.getServerEnvironment().getServerConfigurationFile().getMainFile().getName();
+            message = ServerLogger.AS_ROOT_LOGGER.serverConfigFileInUse(serverConfig);
+        }
+        bootstrapListener.printBootStatistics(message);
```

Comment:
This is a more disruptive change and we should avoid it.

Notice `bootstrapListener.printBootStatistics()` do more things besides just printing the boot statistics message, for example, this method marks as done the Future that returns the service container:

https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/BootstrapListener.java#L91

Moving the invocation of this method to the `postBoot()` will make the Future will return a value after the process transitions to RUNNING. That could have consequences for any thread waiting for the Future. The current behavior is the Future value will be returned before transitioning to RUNNING.

We should avoid changes like this one if we don't have any strong argument to do it. So far, the only motivation is to change the order of the messages, which is weak to support the change.

`postBoot()` looks adequate to fix the issue, but maybe you have to split the `.printBootStatistics()` into two methods (phases to print the message), one method to prepare the message (and to return to the future and do the same it is doing now) and the other to print it on the `postBoot()`.

You should also take a look at the Domain Mode variant since the secondary hosts could be also affected by this unexpected order of messages

Reply:
I am sorry that it took me so long. 
I force-pushed changes.

Reply:
Hi @khermano , sure, don't worry, it is ok if it took time.